lib/routes/api.js

```diff
@@ -15,59 +15,73 @@ const api = require('../api');
 const routes = application => {
   const app = application;
 
-  return api.types(app).then(types => {
-    /*
-     *  API for all Content
-     */
-    // TODO: Tests
-    // TODO: Return error if no items are found
-    app.get('/api', (req, res, next) => {
-      api.all(req.query).then(formatted => {
-        res.json(formatted);
-      }).catch(err => {
-        next(err);
-      });
-    });
+  const types = api.types(app);
 
-    /*
-     *  API for all Content Types
-     */
-    // TODO: Tests
-    // TODO: Return error if no items are found
-    app.get('/api/types', (req, res) => {
-      const formatted = api.content(req.query, types);
+  /*
+   *  API for all Content
+   */
+  // TODO: Tests
+  // TODO: Return error if no items are found
+  app.get('/api', (req, res, next) => {
+    api.all(req.query).then(formatted => {
+      res.json(formatted);
+    }).catch(err => {
+      next(err);
+    });
+  });
 
+  /*
+   *  API for all Content Types
+   */
+  // TODO: Tests
+  // TODO: Return error if no items are found
+  app.get('/api/types', (req, res, next) => {
+    api.content(req.query, types).then(formatted => {
       res.json(formatted);
+    }).catch(err => {
+      next(err);
     });
+  });
 
-    /*
-     *  API for all Content of a given Content Type
-     */
-    // TODO: Tests
-    // TODO: Return error if no items are found
-    app.get('/api/types/:type', (req, res, next) => {
-      api.ofType(req.query, req.params.type).then(formatted => {
-        res.json(formatted);
-      }).catch(err => {
-        next(err);
-      });
+  /*
+   *  API for all Content of a given Content Type
+   */
+  // TODO: Tests
+  // TODO: Return error if no items are found
+  app.get('/api/types/:type', (req, res, next) => {
+    api.ofType(req.query, req.params.type).then(formatted => {
+      res.json(formatted);
+    }).catch(err => {
+      next(err);
     });
+  });
+
+  /*
+   *  Specific Piece of Content
+   */
+  // TODO: Tests
+  // TODO: Return error if no items are found
+  app.get('/api/types/:type/:id', (req, res, next) => {
+    const type = app.get('content-types').find(item => {
+      if (item.id === req.params.type.toLowerCase()) {
+        return true;
+      }
 
-    /*
-     *  Specific Piece of Content
-     */
-    // TODO: Tests
-    // TODO: Return error if no items are found
-    app.get('/api/types/:type/:id', (req, res, next) => {
-      api.one(req.query, req.params.id).then(formatted => {
-        res.json(formatted);
-      }).catch(err => {
-        next(err);
-      });
+      return false;
     });
 
-    return app;
+    if (types.length === 0) {
```

Comment:
perhaps this stuff should be in middleware instead to keep the route as just a promise and to share the test of the url params across routes.